DearPyGui/src/core/PythonCommands/mvTableInterface.cpp

```diff
@@ -163,6 +168,30 @@ namespace Marvel{
 
 		return GetPyNone();
 	}
+	
+	PyObject* hide_headers(PyObject* self, PyObject* args, PyObject* kwargs)
+	{
+		const char* table;
+		bool hide;
+
+		if (!(*mvApp::GetApp()->getParsers())["hide_headers"].parse(args, kwargs, __FUNCTION__, &table, &hide))
+			return GetPyNone();
+
+		auto item = mvApp::GetApp()->getItemRegistry().getItem(table);
```

Comment:
You need to add the lock guard here, like this:

```cpp
std::lock_guard<std::mutex> lk(mvApp::GetApp()->GetApp()->getMutex());
auto item = mvApp::GetApp()->getItemRegistry().getItem(table);
```

This not only protects the item registry from concurrent access but also acts as a synchronization device between the render thread and callback thread.

Reply:
Use configure_itemi("table_id", hide_headers=True) instead of a new function of hide_headers("table_id",True). So we don't use mvApp::GetApp()->getItemRegistry().getItem(table); anymore.